### PR TITLE
moving scraper functions for better organisation

### PR DESCRIPTION
**Pull request by exitflynn**
possible locations are `net.scraper_utils`, `util.net` or outside the scraper class but still in `net.scraper`

### CODE REVIEW

File: sunpy/net/scraper_utils.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+from sunpy.extern.parse import parse
+from sunpy.time import TimeRange
+
+TIME_QUANTITIES = {'day': timedelta(days=1),
+                   'hour': timedelta(hours=1),
+                   'minute': timedelta(minutes=1),
+                   'second': timedelta(seconds=1),
+                   'millisecond': timedelta(milliseconds=1)}
+
+def check_timerange(pattern, url, timerange):
```
Comment (nabobalis) [inline sunpy/net/scraper_utils.py:15]: You will want to move any unit tests for these into a new file as well.

File: sunpy/net/scraper_utils.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        exdict['year'] = 2000 + exdict['year']
+    if 'month' not in exdict:
+                if 'month_name' in exdict:
+                    exdict['month'] = datetime.strptime(exdict['month_name'], '%B').month
+                elif 'month_name_abbr' in exdict:
+                    exdict['month'] = datetime.strptime(exdict['month_name_abbr'], '%b').month
+    tr = get_timerange_from_exdict(exdict)
+    return tr.intersects(timerange)
+
+def smallerPattern(directoryPattern):
```
Comment (nabobalis) [inline sunpy/net/scraper_utils.py]: We should rename this to follow the Python naming style.

File: sunpy/net/scraper_utils.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    Obtain the smaller time step for the given pattern.
+    """
+    try:
```
Comment (nabobalis) [inline sunpy/net/scraper_utils.py]: Does this need to be in a try block?

Comment (exitflynn) [reply sunpy/net/scraper_utils.py]: I don't think so, I'll revert it. I traced it back to the [second](https://github.com/sunpy/sunpy/commit/4469744d7718c97857b09d74b4ddb5a33055f7d5) scraper commit ever :P

File: sunpy/net/scraper_utils.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    ----------
+    datetime : `datetime.datetime` or `astropy.time.Time`
+        The date to floor
+    timestep : `dateutil.relativedelta.relativedelta`
+        The smallest time step to floor
+    Returns
+    -------
+    `datetime.datetime`
+        The time floored at the given time step
+
```
Comment (nabobalis) [inline sunpy/net/scraper_utils.py]:
```suggestion
```

**Pull request merged**